node/services/service.go

```diff
@@ -0,0 +1,37 @@
+package service
+
+import (
+	"context"
+
+	"github.com/celestiaorg/celestia-node/service/block"
+
+	ipld "github.com/ipfs/go-ipld-format"
+	"go.uber.org/fx"
+)
+
+// Components wraps services creation
+func Components() fx.Option {
+	return fx.Options(
+		fx.Provide(func(params blockServiceParams) *block.Service {
+			service := block.NewBlockService(params.Fetcher, params.Store)
+			params.Lc.Append(fx.Hook{
+				OnStart: func(context.Context) error {
+					return service.Start(params.Ctx)
+				},
+				OnStop: func(context.Context) error {
+					return service.Stop(params.Ctx)
```

Comment:
@vgonkivs, also, you should pass contexts there not from params, but from respective hooks.

Reply:
So this way you can also remove Ctx from `blockServiceParams`

Reply:
Ok, thanks. I thought it should correspond to one exact context. Will change it.

Reply:
@Wondertan , fixed.